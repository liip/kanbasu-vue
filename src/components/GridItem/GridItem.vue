<template>
  <div :class="classNames">
    <slot></slot>
  </div>
</template>

<script>
export default {
  name: 'grid-item',

  props: {
    /**
     * `{ default: '1/2', md: '1/3', lg: '1/4', â€¦ }`
     */
    widths: {
      type: Object,
      required: true,
    },
  },

  computed: {
    classNames() {
      const classes = ['grid__item'];

      Object.keys(this.widths).forEach(breakpoint => {
        let ratio = this.widths[breakpoint];

        classes.push(
          `${breakpoint !== 'default' ? `${breakpoint}-` : ''}w-${ratio}`
        );
      });

      return classes;
    },
  },
};
</script>
