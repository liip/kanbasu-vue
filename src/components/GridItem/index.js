import GridItem from './GridItem.vue';

export default GridItem;
