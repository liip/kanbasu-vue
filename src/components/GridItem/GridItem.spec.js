import { shallowMount } from '@vue/test-utils';
import GridItem from './GridItem';

describe('GridItem.vue', () => {
  it('renders', () => {
    const wrapper = shallowMount(GridItem, {
      propsData: {
        widths: {
          default: '1/2',
          sm: '1/3',
          md: '1/4',
        },
      },
    });
    expect(wrapper).toMatchSnapshot();
  });
});
