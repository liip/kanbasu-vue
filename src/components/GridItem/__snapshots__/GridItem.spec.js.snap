// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridItem.vue renders 1`] = `<div class="grid__item w-1/2 sm-w-1/3 md-w-1/4"></div>`;
